src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -895,16 +913,21 @@ internal SqlCollation Collation
             set => _collation = value;
         }
 
+        private bool HasFlag(SqlParameterFlags flag)
+        {
+            return (_flags & flag) != 0;
+        }
+
```

Comment:
I can but why? We can't make it generic over the enum type because you can't constrain to `enum` (only `Enum` which is a ref type and causes the box I'm trying to avoid)

Reply:
I didn't mean an extension for any enum type in general, just for `SqlParameterFlags`. 
At least, it would be more readable
```
_flags.HasFlag(SqlParameterFlags.something);
_flags = _flags.SetFlag(SqlParameterFlags.something, false);
_flags = _flags.Set(SqlParameterFlags.something);
_flags = _flags.Clear(SqlParameterFlags.something);
```
It's not a required change. You can skip this change If you don't see any benefit on it.
```
static class Extension
{
    public static bool HasFlag(this SqlParameterFlags flags, SqlParameterFlags flag)
    {
        return (flags & flag) != 0;
    }
}
```

Reply:
I don't see it as beneficial. It adds a new type which we'll never need to have anything else on and we also can't use HasFlag as an extension method because instance methods override extensions methods in resolution semantics so we'd need to call it HasFlagEx or something else. There just doesn't seem to be much upside to taking those methods out of the class. It's also worth noting that as they're written now there is a high chance that they will be inlined.